WcaOnRails/app/views/competitions/_index_form.html.erb

```diff
@@ -22,24 +22,52 @@
 </div>
 
 <div class="form-group">
-  <%= label_tag(:years) %>
-  <%= select_tag(:years, options_for_select(@years, selected: params[:years], disabled: ""), class: "form-control") %>
-</div>
-
-<div class="form-group">
   <%= label_tag("Name, City or Venue") %>
   <%= text_field_tag "search", params[:search], class: "form-control" %>
 </div>
 
 <div class="form-group">
-  <label>&nbsp;</label>
-  <div class="btn-group">
-    <%= button_tag(type: 'submit', name: "display", value: "List", class: "btn btn-info") do %>
-      <i class="glyphicon glyphicon-list"></i> List
-    <% end %>
-
-    <%= button_tag(type: 'submit', name: "display", value: "Map", class: "btn btn-primary") do %>
-      <i class="glyphicon glyphicon-map-marker"></i> Map
-    <% end %>
+  <%= label_tag(:state) %>
+  <div id="state" class="btn-group" data-toggle="buttons">
+    <label id="present" class="btn btn-primary">
+      <%= radio_button_tag "state", "present" %>
+      <span class="caption">Present</span>
+    </label>
+    <ul class="dropdown-menu years">
+      <%= hidden_field_tag "year", params[:year] %>
+      <% @years.each do |year| %>
+        <li class="year"><%= link_to year.to_s.capitalize, "#", data: { year: year } %></li>
+      <% end %>
+    </ul>
+    <label id="past" class="btn btn-primary" data-toggle="dropdown">
+      <%= radio_button_tag "state", "past" %>
+      <span class="caption">Past <%= "from #{params[:year]}" if @past_selected %></span>
+    </label>
   </div>
 </div>
+
+<script>
+  $('#<%= params[:state] %>').button('toggle');
```

Comment:
Good catch. Actually #button method is not something danger I think. Anyways probably we should use it in more explicit way that there's always 'present' or 'past'.